node/core/chain-selection/src/tree.rs

```diff
@@ -394,6 +393,47 @@ fn apply_reversions(
 	Ok(())
 }
 
+/// Marks a single block as explicitly reverted. Then propegates viability updates
+/// to all its children. This is triggered when the disputes subsystem signals that
+/// a dispute has concluded against a candidate.
+pub(crate) fn apply_single_reversion(
```

Comment:
off by one error. You are using the reverted block, but we should be using the reverted to block.

Reply:
Maybe we can then also use `apply_ancestor_revisions` again? Ideally we have mostly the same code path to reversion logs, with the only difference that the reversion info is not coming from logs, but from the dispute-coordinator.

Reply:
I think the resulting code would be less clear if we tried to tuck both these reversion cases into one function. This is mainly because of differences in necessary parameters.

`apply_ancestor_reversions` needs reversion logs and info about the imported leaf they came from. 

`apply_single_reversion` doesn't need either of those pieces of information, and indeed couldn't provide them since there is no new imported leaf involved. It instead needs direct information about a single block to revert.

Reply:
> off by one error. You are using the reverted block, but we should be using the reverted to block.

The underlying tree structure for chain selection has a concept of an `explicitly_reverted` block, but doesn't have a concept for reverting to a particular block. It instead adds the block we'd think of as `reverted_to` back to the set of viable leaves as a consequence of `propagate_viability_update`. 

If you think the framing of reverting to a block is more intuitive, then we can definitely reframe things. But the information chain selection needs from the dispute coordinator is the hash and block number of the block to be marked as reverted, not it's parent.

Reply:
All good. My mistake, thanks @tdimitrov for highlighting what I missed.